doc/concepts/concepts.rst

```diff
@@ -1,13 +1,228 @@
+========
```

Comment:
Would it be useful to break some of this up into separate sub-pages in the concepts section?  That'll make it easier to expand this in the future.

Reply:
Yeah, that makes sense to me.